addons/hr_expense/models/hr_expense.py

```diff
@@ -376,6 +376,9 @@ def write(self, vals):
         if 'reference' in vals:
             if any(not expense.is_ref_editable for expense in self):
                 raise UserError(_('You are not authorized to edit the reference of this expense report.'))
+        if (not self.user_has_groups('hr_expense.group_hr_expense_manager') and vals.get('state') != 'submit' and
+            any(expense.state == 'draft' for expense in self)):
+                raise UserError(_("You are not authorized to bypass the expense validation process."))
```

Comment:
The issue with the current validation process is that it can be bypassed by an RPC call using the "write" method on a draft and writing the state to something else like "approved". 

So in my opinion in doesn't break the validation process. it just has to ensure that the expense goes from draft to submit and not from draft to something else (unless you have full right access).